integreat_cms/static/src/js/pois/poi-map.ts

```diff
@@ -1,165 +1,242 @@
+            zoom: 15,
+        });
+    } else {
+        // Else, remove the marker from the map
+        marker.remove();
+        // And show a bounding box of the region
+        map.fitBounds(JSON.parse(map.getContainer().dataset.boundingBox) as LngLatBoundsLike);
+    }
+
+    updateHelpText(coordinates);
+}
+
+// Adjust help text below the map
+function updateHelpText(coordinates: LngLat) {
+    const setMapPositionText = document.getElementById("set_map_position_text");
+    const changeMapPositionText = document.getElementById("change_map_position_text");
+    if (coordinates) {
+        setMapPositionText.classList.add("hidden");
+        changeMapPositionText.classList.remove("hidden");
+    } else {
+        setMapPositionText.classList.remove("hidden");
+        changeMapPositionText.classList.add("hidden");
+    }
 }
 
-async function drag_and_drop(lngLat: maplibregl.LngLat) {
-    document.getElementById("drag_and_drop_change").classList.remove("hidden");
+// Fetch the address at a given location
+async function fetchAddress(marker: Marker) {
+    // Show loading spinner
+    document.getElementById("address_loading").classList.remove("hidden");
+    // Hide error in case it was shown before
+    document.getElementById("no_address_found").classList.add("hidden");
+
+    // Get the new coordinates of the marker
+    const coordinates = marker.getLngLat();
+
+    // Show new coordinates
+    document.getElementById("updated_latitude").textContent = coordinates.lat.toString();
+    document.getElementById("updated_longitude").textContent = coordinates.lng.toString();
+    document.getElementById("update_position_from_map_marker").classList.remove("hidden");
 
+    // Fetch the address at this coordinates
     const response = await fetch(document.getElementById("map").dataset.url, {
         method: "POST",
         headers: {
             "X-CSRFToken": getCsrfToken(),
         },
         body: JSON.stringify({
-            lng: lngLat.lng,
-            lat: lngLat.lat,
+            longitude: coordinates.lng,
+            latitude: coordinates.lat,
         }),
     });
 
-    const data = await response.json();
-
-    let error = document.getElementById("nominatim-error");
-    if (response.status != 200) {
-        error.textContent = data.error;
-        error.classList.remove("hidden");
-        return;
-    }
-
-    let address = construct_address(data.number, data.street, data.postcode, data.city, data.country);
-    let coordinates = "Latitude: " + lngLat.lat + ", Longitude: " + lngLat.lng;
-
-    document.getElementById("address_found").textContent = address;
-    document.getElementById("coordinates_found").textContent = coordinates;
-
-    let street_number = data.street + " " + data.number;
-    document.getElementById("confirm_map_input").addEventListener("click", () => {
-        use_map_pin_adresse(street_number, data.postcode, data.city, data.country, lngLat.lat, lngLat.lng);
-    });
```

Comment:
If you set the event listener each time the coordinates change, you pile up a bunch of event handlers and if you use the feature multiple times subsequently, the function will run too often... Instead, we should set this event handler on `load` and think about another way to pass the new values to the update function (I did it by splitting the fields into separate `<span>` elements which can just be retrieved when the new position is confirmed).